src/datachain/query/dataset.py

```diff
@@ -682,7 +675,15 @@ def q(*columns):
             names = {c.name for c in columns}
             # Columns for the generated table.
             cols = [c for c in udf_table_cols if c.name in names]
-            return sqlalchemy.select(*cols).select_from(udf_table_query)
+            res = sqlalchemy.select(*cols).select_from(udf_table_query)
+
+            if query._order_by_clauses:
```

Comment:
qq - how come you are switching from `if not query._order_by_clauses:` to `if query._order_by_clauses:`?

Reply:
To be honest I am not sure why it was `not` there, may be it is because of previous refactoring attempts.